source/ShipyardPanel.cpp

```diff
@@ -240,6 +252,9 @@ void ShipyardPanel::Buy(bool onlyOwned)
 	if(licenseCost < 0)
 		return;
 
+	if(!shipyard.Has(selectedShip))
+		player.RemoveShipStock(selectedShip);
+
```

Comment:
This is not needed, and it removes a ship from the stock even if the player cancels the purchase at the name dialog.
```suggestion
```